README.md

```diff
@@ -1,335 +1,446 @@
 # apt
 
+#### Table of Contents
+
+1. [Overview](#overview)
+2. [Module Description - What the module does and why it is useful](#module-description)
+3. [Setup - The basics of getting started with [apt]](#setup)
+    * [What apt affects](#what-apt-affects)
+    * [Beginning with apt](#beginning-with-apt)
+4. [Usage - Configuration options and additional functionality](#usage)
+5. [Reference - An under-the-hood peek at what the module is doing and how](#reference)
+    * [Classes](#classes)
+    * [Defines](#defines)
+    * [Types](#types)
+    * [Facts](#facts)
+6. [Limitations - OS compatibility, etc.](#limitations)
+7. [Development - Guide for contributing to the module](#development)
+
 ## Overview
 
-The apt module provides a simple interface for managing Apt source, key, and definitions with Puppet.
+The apt module lets you use Puppet to manage Apt sources, keys, and definitions.
 
 ## Module Description
 
-The apt module automates obtaining and installing software packages on \*nix systems.
-
-**Note**: While this module allows the use of short keys, **warnings are thrown if a full fingerprint is not used**, as they pose a serious security issue by opening you up to collision attacks.
+Apt (Advanced Package Tool) is a package manager available on Debian, Ubuntu, and several other operating systems. The apt module provides a series of classes, defines, types, and facts to help you automate Apt package management.
 
 ## Setup
 
-### What apt affects:
+### What apt affects
 
-* Package/service/configuration files for Apt
 * Your system's `sources.list` file and `sources.list.d` directory
 * System repositories
 * Authentication keys
 
+**Note:** Setting the apt module's `purge_sources_list` and `purge_sources_list_d` parameters to 'true' will destroy any existing content that you haven't declared through Puppet. The default for these parameters is 'false'.
+
 ### Beginning with apt
 
-To begin using the apt module with default parameters, declare the class with `include apt`.
+To use the apt module with default parameters, declare the `apt` class.
 
-Any Puppet code that uses anything from the apt module requires that the core apt class be declared.
+~~~
```

Comment:
So why do we have `~~~yaml` in places but not `~~~puppet`? And if it doesn't work on your main doc site, what's the harm in adding it anyway so that it does render nicely on Github?